nixos/lib/make-iso9660-image.sh

```diff
@@ -100,6 +100,7 @@ done
 mkdir -p $out/iso
 
 xorriso="xorriso
+ -boot_image any gpt_disk_guid=$(uuid -v 5 daed2280-b91e-42c0-aed6-82c825ca41f3 $out | tr -d -)
```

Comment:
Where does the `daed2280-b91e-42c0-aed6-82c825ca41f3` come from? Is it chosen arbitrarily? How does it influence the result and how does it work together with `$out`?

I think a comment explaining why we fix the guid and why we chose the parameters like this would be good. Then this should be good to go. You seem to have done sufficient testing.

Reply:
It is an arbitrary constant.

Version-3 and Version-5 UUID's are derived from the combination of a value ($out) and a namespace (this random constant).

The reasoning behind having a namespace is so that 2 independent applications generating a UUID for the same value will generate different UUID's (though the chance of that being a problem seems pretty slim in this particular case).

I'll add a comment.